streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -590,18 +591,18 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
     /**
      * Repartition a stream. This is required on join operations occurring after
      * an operation that changes the key, i.e, selectKey, map(..), flatMap(..).
-     *
-     * @param joined joined control object
-     * @return a new {@link KStreamImpl}
      */
-    private KStreamImpl<K, V> repartitionForJoin(final Joined<K, V, ?> joined) {
-        final Serde<K> repartitionKeySerde = joined.keySerde() != null ? joined.keySerde() : keySerde;
-        final Serde<V> repartitionValueSerde = joined.valueSerde() != null ? joined.valueSerde() : valSerde;
+    private KStreamImpl<K, V> repartitionForJoin(final String repartitionNameOverride,
+                                                 final Serde<K> keySerdeOverride,
+                                                 final Serde<V> valueSerdeOverride) {
+        final String name = repartitionNameOverride != null ? repartitionNameOverride : this.name;
```

Comment:
All the callers seems already handles the `name == null` case so this seems unnecessary.

Reply:
This just simplifies the callers, who would just have to do the same check at the call site: https://github.com/apache/kafka/pull/5741/files#diff-5142e1d4a6410459d6bf6df98828e5afL682 and https://github.com/apache/kafka/pull/5741/files#diff-5142e1d4a6410459d6bf6df98828e5afL703

Reply:
ok, I had removed that check from half the callers, but at Bill's request, I've pushed it back out.